src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -188,3 +189,35 @@ get_outgoing_pools() ->
     [#{report_name => outgoing_pools,
        key => type,
        value => Type} || {Type, _, _, _, _} <- OutgoingPools].
+
+get_xmpp_stanzas_count() ->
+    Hosts = ejabberd_config:get_global_option(hosts),
+    StanzaTypes = [xmppMessageSent, xmppMessageReceived, xmppIqSent,
+                   xmppIqReceived, xmppPresenceSent, xmppPresenceReceived],
+    NewCount = [count_stanzas(Hosts, StanzaType) || StanzaType <- StanzaTypes],
+    PrevReport = mongoose_system_metrics_file:read(),
+    StanzasCount = calculate_stanza_rate(PrevReport, NewCount),
+    [#{report_name => StanzaType,
+       key => Total,
+       value => Increment} || {StanzaType, Total, Increment} <- StanzasCount].
+
+count_stanzas(Hosts, StanzaType) ->
+    FullExometerResults = [exometer:get_value([Host, StanzaType]) || Host <- Hosts],
```

Comment:
As far as I know even if MongooseIM is configured to have all metrics as global, they have the same level. It's for instance `[Host, xmppMessagesSent]` for per host metrics or `[global, xmppMessagesSent]` if MongooseIM is configured to have all metrics as global. We could use the following exometer API:
```erlang
exometer:get_values(['_', xmppMessageSent])
```
to get all the values despite number of hosts or if metrics are global or not.

Reply:
yep, this would be the best way to get this data